backend/spec/requests/roles_request_spec.rb

```diff
@@ -2,14 +2,40 @@
 
 RSpec.describe 'RolesController', type: :request do
   describe '#list' do
-    let!(:role) { create(:role) }
+    let!(:role_moderator) { create(:role_moderator) }
+    let!(:role_admin) { create(:role_admin) }
+    let!(:role_teacher) { create(:role_teacher) }
+    let!(:role_student) { create(:role_student) }
+    let(:user_admin) { create(:user) }
 
-    before { get api_v1_roles_path }
+    describe 'List all roles when admin' do
```

Comment:
usually specs start with the word `when`
like `when user is admin, list all roles`

Reply:
Done!